testslide/mock_constructor.py

```diff
@@ -157,11 +157,24 @@ def __delete__(self, instance):
         )
 
 
-def _get_mocked_class(original_class, target_class_id, callable_mock):
+def _wrap_type_validation(callable_mock, templates):
+    def callable_mock_with_type_validation(*args, **kwargs):
+        for template in templates:
+            _validate_function_signature(template, args, kwargs)
+        return callable_mock(*args, **kwargs)
+
+    return callable_mock_with_type_validation
+
+
+def _get_mocked_class(original_class, target_class_id, callable_mock, type_validation):
     if target_class_id in _target_class_id_by_original_class_id:
         raise RuntimeError("Can not mock the same class at two different modules!")
     else:
         _target_class_id_by_original_class_id[id(original_class)] = target_class_id
+
+    original_class_new = original_class.__new__
```

Comment:
I would have forgotten about new xd

Reply:
Turns out, it is there only for being future proof, as `mock_constructor()` currently can't handle classes that define their own `__new__`: #155.